tests/src/python/test_provider_postgres.py

```diff
@@ -397,7 +397,24 @@ def testTransactionNotDirty(self):
         ft1 = vl.getFeatures('pk=1')
         self.assertFalse(ft1.nextFeature(f))
 
-        p.setAutoTransaction(False)
+    def testTransactionTuple(self):
+        # create a vector layer based on postgres
+        vl = QgsVectorLayer(self.dbconn + ' sslmode=disable key=\'pk\' srid=4326 type=POLYGON table="qgis_test"."some_poly_data" (geom) sql=', 'test', 'postgres')
+        self.assertTrue(vl.isValid())
+
+        # prepare a project with transactions enabled
+        p = QgsProject()
+        p.setAutoTransaction(True)
+        p.addMapLayers([vl])
+        vl.startEditing()
+
+        # execute a query which returns a tuple
+        tr = vl.dataProvider().transaction()
+        sql = "select * from qgis_test.some_poly_data"
+        self.assertTrue(tr.executeSql(sql, False)[0])
```

Comment:
Can we also test that the data returned is as expected?

Reply:
The data is not returned by `executeSql`, only the status code and an error message.

Reply:
How's executing a select useful in such a scenario?
And (assuming it is) do we need to consider already an API for future implementations that return tables/iterators? Not saying we need to implement it right now, but we should leave room for a nice API.

Reply:
> How's executing a select useful in such a scenario?

In this particular case, none. It was just for the test (check that there's no error when the database is returning a tuple).

@haubourg Can you share the kind of SQL requests you're using with `executeSql`?

Reply:
@m-kuhn Yeah, we've beed discussing that with  @vmora when working on the undo-redo in transaction group. 
Being able to really handle tuples inside the same transaction groups would be useful, when only psyco-pg allows that currently but in a different connection. 
We fear not to have strong enough shoulders to open such a task, and more than that, we don't have any concrete use case.  

By now, our only excuse for allowing tuple is that we could not find another way to have PG not return a tuple when  executing a stored procedure. We off course tested that https://stackoverflow.com/questions/1343954/how-do-i-write-a-postgres-stored-procedure-that-doesnt-return-anything with no success, this was still considered as a tuple by QGIS.

Reply:
Fair enough, seems to be a PG limitation also according to https://stackoverflow.com/a/18325437/2319028.

Answering my previous question myself, I guess we could introduce a `select` statement returning anything appropriate (read: iterator) in the future, so should be fine API wise.